main.go

```diff
@@ -436,10 +449,22 @@ See also https://kairos.io/docs/installation/qrcode/ for documentation.
 This command is meant to be used from the boot GRUB menu, but can be started manually`,
 		Aliases: []string{"i"},
 		Before: func(c *cli.Context) error {
+			if err := validateSource(c.String("source")); err != nil {
+				return err
+			}
+
 			return checkRoot()
 		},
+		Flags: []cli.Flag{
+			&cli.StringFlag{
+				Name:  "source",
+				Usage: "Source for upgrade. Composed of `type:address`. Accepts `file:`,`dir:` or `oci:` for the type of source.\nFor example `file:/var/share/myimage.tar`, `dir:/tmp/extracted` or `oci:repo/image:tag`",
```

Comment:
Maybe I'm mistaken to ask this but it feels to me like this should be a proper URI instead of our own way of doing this ... but maybe this is a better UX, not sure

Reply:
actually, the comment might be wrong. [We use `url.Parse` to parse this value](https://github.com/kairos-io/kairos-agent/blob/1170576dcce4018a3dffddf500295c3ac3a64a46/pkg/types/v1/common.go#L93), so `oci://` is probably the right format.

Reply:
it is, as an URI just defines the protocal:address and we want this to be generic enough.

Reply:
@Itxaka you mean "it is" as in "it is wrong" ? Should I change the text to have forward slashes too? e.g. `oci://`, `file://`, `dir://` etc?

Reply:
it is valid and the way of doing it, otherwise the value cannot be obtained if we use `protocol://address`

Dont ask me why, it just doesnt work with the `://` properly. Maybe our parsing is incorrect or we are using it wrong: https://goplay.tools/snippet/AMNXoOVJr_6

Reply:
In any case, I think this would be out of scope for this patch.

Reply:
Yes, I agress, it's out of scope.